libs/output-mapping/src/Writer/Table/TableConfigurationValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Writer\Table;
+
+use InvalidArgumentException;
+use Keboola\OutputMapping\Exception\InvalidOutputException;
+use Keboola\OutputMapping\Mapping\MappingFromRawConfigurationAndPhysicalDataWithManifest;
+use Keboola\OutputMapping\Writer\Helper\RestrictedColumnsHelper;
+use Keboola\OutputMapping\Writer\Table\Strategy\SqlWorkspaceTableStrategyNew;
+
+class TableConfigurationValidator
+{
+    public function validate(
+        StrategyInterfaceNew $strategy,
+        MappingFromRawConfigurationAndPhysicalDataWithManifest $source,
+        array $config,
+    ): array {
+        if (!$strategy instanceof SqlWorkspaceTableStrategyNew) {
+            try {
+                RestrictedColumnsHelper::validateRestrictedColumnsInConfig(
+                    $config['columns'],
+                    $config['column_metadata'],
+                );
+            } catch (InvalidOutputException $e) {
+                throw new InvalidOutputException(
+                    sprintf(
+                        'Failed to process mapping for table %s: %s',
+                        $source->getSourceName(),
+                        $e->getMessage(),
+                    ),
+                    0,
+                    $e,
+                );
+            }
+        }
+
+        $hasColumns = $config['columns'];
```

Comment:
tak $hasColumns by aspoň mohlo být bool :)

Reply:
no spíš jsem to dal úplně pryč - používá se na jednom místě